### ISSUE TITLE
Preprocessing optimisations

### ISSUE THREAD

**Issue by wokalski** [wokalski/Diff.swift#5]
https://neil.fraser.name/writing/diff/

**Comment by nitrogenice** [comment]
Hi, may I contribute?

**Comment by wokalski** [comment]
@nitrogenice you are more than welcome

**Comment by nitrogenice** [comment]
@wokalski Great! I guess I will start with some really simple optimizations. Have you implemented any of the optimizations described in the link above? I have not noticed any yet.

**Comment by wokalski** [comment]
No. This is basically where the whole diff algorithm [begins](https://github.com/wokalski/Diff.swift/blob/master/Sources/Diff.swift#L177).

I just check some basic preconditions
